src/webapp/reports/autogenerated-forms/GridFormViewModel.ts

```diff
@@ -74,15 +96,39 @@ export class GridViewModel {
             })
             .value();
 
+        const dataElementsByTotal = _(section.calculateTotals)
+            .groupBy(item => item?.totalDeCode)
+            .map((group, totalColumn) => ({
+                totalColumn,
+                dataElements: group.map(item => _.findKey(section.calculateTotals, obj => obj === item)),
+            }))
+            .value();
+
         const rows = subsections.map(subsection => {
             const firstDataElement = _(subsection.dataElements).first();
             const indicator = getIndicatorRelatedToDataElement(section.indicators, firstDataElement?.code || "");
             const items = columns.map(column => {
                 const dataElement = subsection.dataElements.find(de => de.name === column.name);
+                const deCalculateTotal =
+                    section.calculateTotals && dataElement?.code
+                        ? section.calculateTotals[dataElement.code]
+                        : undefined;
+
+                const parentTotal = deCalculateTotal
+                    ? dataElements.find(de => de.code === deCalculateTotal?.totalDeCode)
+                    : undefined;
+
+                const dataElementsInTotalColumn = dataElementsByTotal.find(x => x.totalColumn === parentTotal?.code);
+                const columnDataElements = dataElements.filter(de =>
+                    dataElementsInTotalColumn?.dataElements.includes(de.code)
+                );
 
                 return {
-                    column,
-                    dataElement,
+                    column: column,
+                    columnTotal: deCalculateTotal?.totalDeCode ? parentTotal : undefined,
```

Comment:
Since `parentTotal` is calculated from `deCalculateTotal` I think we can pass it directly, right?

```tsx
columnTotal: parentTotal
```